### ISSUE TITLE
Support for other later version of SQL Server 18 ODBC?

### ISSUE THREAD

**Issue by mkarikari** [Esri/arcgis-powershell-dsc#267]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Module Version

* 3.0.0

### Affected Resource(s)

* SQLClient

### Configuration Files

```
        "SQLServerClient": [
            {
                "Name": "Microsoft ODBC Driver 17 for SQL Server",
                "InstallerPath": "\\test\\gis\\software\\ancillary\\msodbcsql_1742.msi",
                "ProductId": "72AFAF21-33FB-45A5-9468-A9EC07427F82",
                "Arguments": " /quiet /qn IACCEPTMSODBCSQLLICENSETERMS=YES ADDLOCAL=ALL"
            }
        ]
```

### Expected Behavior
Does the SQLServerClient support later version of the SQL Driver or is 17 a hard requirement at this point? I was having issue finding the productID info for v18 so could move forward with the test.

**Comment by mkarikari** [comment]
Also, when I do run with the latest version of 17 client I receive the following error

![image](https://user-images.githubusercontent.com/19801488/81926503-0a6e3980-95b0-11ea-949c-13a6ff7c5755.png)
Im not sure if im getting that error because Im using the 17.5 vs 17.4.1, but if so would happen know where I can get the exact version that is prereq as Im having trouble finding that older version

**Comment by cameronkroeker** [comment]
Hello @mkarikari ,

Doesn't look like v18 has been added to the supported list as of yet:

https://desktop.arcgis.com/en/system-requirements/latest/database-requirements-sqlserver.htm#ESRI_SECTION1_B75DB4BEF4BF40348621F378FB8F9FD8

Additionally, 17.5.2 should work. I recommend checking the `"ProductId"` parameter to ensure it matches. Each version is going to have a unique Product ID. For example, the product ID for 17.5.2 is:
```
IdentifyingNumber                      Name
-----------------                      ----
{12DC69AF-787B-4D76-B69D-2716DACA79FB} Microsoft ODBC Driver 17 for SQL Server
```
I was able to obtain the product ID by first downloading and manually installing from:
https://docs.microsoft.com/en-us/sql/connect/odbc/download-odbc-driver-for-sql-server?view=sql-server-ver15#download-for-windows

Then in PowerShell ran the following command:

```get-wmiobject Win32_Product | Format-Table IdentifyingNumber, Name, LocalPackage -AutoSize```

The IdentifyingNumber is the Product ID.
